### PR TITLE
Fee discounts should use the discount currency rather than the base fee currency

### LINKED ISSUE #2371: Overview page: discount volume rendering is not respecting the volume currency
In 884-gh-Automattic/woocommerce-payments-server we introduced discount volume currency (note in account cache, under 'discount' fees section the 'volume_currency' property). 

Currently, the volume widget is using base fee currency, and we need to use the new property instead.

### PR DESCRIPTION
**Pull request by aprea**
Fixes #2371

#### Changes proposed in this Pull Request

Fee discounts displayed on the overview page should use the discount currency rather than the base fee currency.

e.g. You have a store in the UK with GBP set as the store currency. You have a fee discount applied to your account for $25,000 USD. This discount should be displayed as USD on the overview page rather than GBP.

| Before | After |
|-|-|
|Screen Shot 2021-07-20 at 2 18 54 pm|Screen Shot 2021-07-20 at 2 15 12 pm|

#### Testing instructions

* Reonboard your store a non-USD merchant.
* Assign a promo to your test account via manual DB query (I used the `wcpay-promo-2021-6-incentive-2` promo)
* Refresh the account cache on the client-side in WCPay Dev and ensure **Payments → Overview** shows the progress bar and discount description representing the volume in USD.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/components/account-status/account-fees/index.js [hunk 1]
```diff
@@ -51,11 +51,15 @@ const AccountFees = ( props ) => {
 			{ feeDescription }
 			<ExpirationBar
 				feeData={ currentFee }
-				currencyCode={ currencyCode }
+				currencyCode={
```
Comment (kalessil) [inline client/components/account-status/account-fees/index.js:54]: We are passing `currentFee` into ExpirationBar, can we drop `currencyCode` property and move the logic into `ExpirationBar`?

File: client/components/account-status/account-fees/index.js [hunk 2]
```diff
@@ -51,11 +51,15 @@ const AccountFees = ( props ) => {
...
+					currentFee.volume_currency ?? currentFee.currency
+				}
 			/>
 			<ExpirationDescription
 				feeData={ currentFee }
-				currencyCode={ currencyCode }
+				currencyCode={
+					currentFee.volume_currency ?? currentFee.currency
+				}
```
Comment (kalessil) [inline client/components/account-status/account-fees/index.js:62]: Same here: can we move the logic into `ExpirationDescription`?

**Review by kalessil — CHANGES_REQUESTED** [review]

File: client/components/account-status/account-fees/index.js (hunk 1 above)
Comment (aprea) [reply client/components/account-status/account-fees/index.js]: Addressed in 1921d85

File: client/components/account-status/account-fees/index.js (hunk 2 above)
Comment (aprea) [reply client/components/account-status/account-fees/index.js]: Addressed in 1921d85

**Review by htdat — COMMENTED** [review]
I tested with a Spain account: 

Markup on 2021-07-21 at 12:08:40

The only concern is that it will be very confusing for AU, CA, NZ users, or any country that uses the same currency symbol, e.g. `$`. It looks this issue will be addressed here https://github.com/Automattic/woocommerce-payments/pull/2171

I am not sure the best way to handle this situation. IMHO, we can go ahead, always display the currency code, and leave a TODO here? And then make a note in the PR above to remove our handling when the PR is merged.

**Comment by kalessil** [comment]
@htdat : I think with this specific PR we are good - we rendering promo volume currency properly. In ideal world we will not assign USD-promo fee to ES-merchant =)

**Review by kalessil — APPROVED** [review]
LGTM: :shipit: !

**Pull request merged**